### PR TITLE
[FrameworkBundle] add "run" method to the MicroKernelTrait

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | TODO

One user commented on the MicroKernel [article](https://symfony.com/blog/new-in-symfony-5-1-improved-microkernel) published today, about the code verbosity at the end of the example shown:

```php
// ... 

$app = new App('dev', true);
$request = Request::createFromGlobals();
$response = $app->handle($request);
$response->send();
$app->terminate($request, $response);
```

Taking into account also this code doesn't change normally between apps (almost never) I think he's right and this code can be moved to a new method, making the single-file app API more concise.

Inspired on [Silex](https://silex.symfony.com) this is what I propose:
```php
// ...

App::run('dev', true);
```

In a nutshell, this is how it would look after these changes:
```php
// index.php (single-file app)

require __DIR__.'/../vendor/autoload.php';

class App extends Kernel
{
    use MicroKernelTrait;

    public function registerBundles(): iterable
    {
        yield new Symfony\Bundle\FrameworkBundle\FrameworkBundle();
    }

    protected function configureRoutes(RoutingConfigurator $routes)
    {
        $routes->import(__CLASS__, 'annotation'); // require doctrine/annotations
    }

    protected function configureContainer(ContainerConfigurator $c)
    {
        $c->extension('framework', ['secret' => '12345']);
    }

    /**
     * @Route("/")
     */
    public function index()
    {
        return new Response('Hello World');
    }
}

App::run('dev', true);
```
What do you think?

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php [hunk 1]
```diff
@@ -181,4 +183,19 @@ public function loadRoutes(LoaderInterface $loader)
 
         return $routes->build();
     }
+
+    public static function run(string $environment, bool $debug): void
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:187]: Another way, by removing the `static` word and go this way:
```php
(new App('dev', true))->run();
```

Comment (yceruto) [reply src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:187]: Btw, I found the current way more readable and concise.

**Comment by fabpot** [comment]
I'm  :-1:

**Comment by ro0NL** [comment]
this effectively vendor locks the front controller by default; it's not worth it IMHO.

**Comment by nicolas-grekas** [comment]
The less core deals with global state the better. Closing as this is objected already and won't happen. Thanks for giving it a try.
Check https://github.com/tchwork/bootstrapper for an alternative approach :)
